index/index.go

```diff
@@ -563,27 +571,32 @@ func (b realByteSlice) Sub(start, end int) ByteSlice {
 }
 
 // NewReader returns a new IndexReader on the given byte slice.
-func NewReader(b ByteSlice) (*Reader, error) {
-	return newReader(b, nil)
+func NewReader(b ByteSlice, version int) (*Reader, error) {
+	return newReader(b, nil, version)
+}
+
+func NewReaderV1(b ByteSlice, c io.Closer, version int) (*Reader, error) {
```

Comment:
Do we need this function with the above definition already doing the same thing? I know I was leaning towards a `NewReaderV1` and `NewReaderV2`, but that doesn't look necessary right now.

Reply:
Also, if its `V1` why is it taking a version ;)?